pytensor/xtensor/math.py

```diff
@@ -134,3 +138,115 @@ def cast(x, dtype):
     if dtype not in _xelemwise_cast_op:
         _xelemwise_cast_op[dtype] = XElemwise(scalar_op=_cast_mapping[dtype])
     return _xelemwise_cast_op[dtype](x)
+
+
+class XDot(XOp):
+    """Matrix multiplication between two XTensorVariables.
+
+    This operation performs matrix multiplication between two tensors, automatically
+    aligning and contracting dimensions. The behavior matches xarray's dot operation.
+
+    Parameters
+    ----------
+    dims : tuple of str
+        The dimensions to contract over. If None, will contract over all matching dimensions.
+    sum_result : bool
+        If True, sum over all remaining axes after contraction (for full contraction, e.g. dims=...).
+    """
+
+    __props__ = ("dims", "sum_result")
```

Comment:
Instead of putting the burden of `sum_resulst` in the XDot Op, combine XDot with Sum in the helper function, since the reduction behavior is already well specified by Sum.

Reply:
yes, that's better